tasks/go_test.py

```diff
@@ -296,6 +298,34 @@ def sanitize_env_vars():
             del os.environ[env]
 
 
+def process_test_result(test_results, junit_tar, flavor, test_washer):
+    if junit_tar:
+        junit_files = []
+        for module_test_result in test_results:
+            if module_test_result.junit_file_path:
+                junit_files.append(module_test_result.junit_file_path)
+
+        produce_junit_tar(junit_files, junit_tar)
+
+    success = process_module_results(flavor=flavor, module_results=test_results)
+
+    if success:
+        print(color_message("All tests passed", "green"))
+        return True
+
+    if test_washer:
+        tw = TestWasher()
+        should_succeed = tw.process_module_results(test_results)
+        if should_succeed:
+            print(
+                color_message("All failing tests are known to be flaky, marking the test job as successful", "orange")
+            )
+            return True
```

Comment:
You do not have any return if `should_succeed` is `False`. I think you can simply get rid of the `else` statement

Reply:
Nice catch!